src/foam/u2/view/TwoFactorAuthenticationVerificationView.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'net.nanopay.sme.ui',
+  name: 'TwoFactorAuthenticationVerificationView',
+  extends: 'foam.u2.View',
+  documentation: `numOfParts stand for length of verification code`,
+  requires: [
+    'foam.core.ArraySlot',
+    'foam.u2.TextField'
+  ],
+
+  css: `
+    ^ {
+      width: 80%;
+      display: flex;
+      flex-direction: row;
+      align-items: center;
+      justify-content: flex-start;
+    }
+    ^ input {
+      border-width: 1px;
+      border-radius: 5px;
+      width: 48px;
+      height: 48px;
+      text-align: center;
+      margin: 8px 14px 8px 0;
+      font-size: 22px;
+    }
+    ^ .wrong-code {
+      border-color: #f91c1c;
+      background-color: #fff6f6;
+    }
+  `,
+  properties: [
+    {
+      class: 'String',
+      name: 'data'
+    },
+    {
+      class: 'Boolean',
+      name: 'incorrectCode',
+    },
+    {
+      class: 'Int',
+      name: 'currentIndex',
+      value: 0,
+      preSet: function(old, nu) {
+        if ( nu > this.numOfParts - 1 ) return this.numOfParts - 1;
+        if ( nu < 1 ) return 0;
+        return nu;
+      }
+    },
+    {
+      class: 'Int',
+      name: 'numOfParts',
+      value: 6,
+    },
+    {
+      class: 'FObjectArray',
+      of: 'foam.u2.Element',
+      name: 'elements'
+    }
+  ],
+  methods: [
+    function initE() {
+      this.SUPER();
+      var self = this;
+      for ( let i = 0; i < this.numOfParts; i++ ) {
+        var isFirstElement = i === 0;
+        let v = this.TextField.create({ onKey: true });
+        v.setAttribute('maxlength', 1);
+        v.setAttribute('autofocus', isFirstElement);
+        v.addClass('input').enableClass('wrong-code', this.incorrectCode$ );
+        v.on('focus', () => {
+          self.currentIndex = i;
+        });
+        v.on('keydown', (e) => {
+          switch ( e.keyCode ) {
+            case 37:
```

Comment:
Add comments to say what characters these keycodes represent.
You can also make these actions and then specify their keyboardShortcuts.